Can't build with Godot 4.2-rc1 and MSVC

### Godot version

4.2-rc1

### Plugin version

HEAD

### System information

Windows 11

### Issue description

I try to build webrtc GDExtension for Godot 4.2-rc1. So Cloned repo and checked out godot-cpp 7cd79e4cc64ce484ba04d1919da7d4cd39aab5ff (matched rc1).
Built with VS2022 and finally got error:

```
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: void __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::pbump(int)" (?pbump@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAXH@Z) already defined in libcpmt.lib(cout.obj)
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::_Pnavail(void)const " (?_Pnavail@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEBA_JXZ) already defined in libcpmt.lib(cout.obj)
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: void __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::_Init(void)" (?_Init@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAXXZ) already defined in libcpmt.lib(cout.obj)
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: void __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::_Init(char * *,char * *,int *,char * *,char * *,int *)" (?_Init@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IEAAXPEAPEAD0PEAH001@Z) already defined in libcpmt.lib(cout.obj)
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: virtual __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::xsgetn(char *,__int64)" (?xsgetn@?$basic_streambuf@DU?$char_traits@D@std@@@std@@MEAA_JPEAD_J@Z) already defined in libcpmt.lib(cout.obj)
msvcprt.lib(MSVCP140.dll) : error LNK2005: "protected: virtual __int64 __cdecl std::basic_streambuf<char,struct std::char_traits<char> >::xsputn(char const *,__int64)" (?xsputn@?$basic_streambuf@DU?$char_traits@D@std@@@std@@MEAA_JPEBD_J@Z) already defined in libcpmt.lib(cout.obj)
   Creating library bin\extension-4.1\webrtc\lib\libwebrtc_native.windows.template_debug.x86_64.lib and object bin\extension-4.1\webrtc\lib\libwebrtc_native.windows.template_debug.x86_64.exp
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library
LINK : error LNK1218: warning treated as error; no output file generated
scons: *** [bin\extension-4.1\webrtc\lib\libwebrtc_native.windows.template_debug.x86_64.dll] Error 4098
scons: building terminated because of errors.
```

### Steps to reproduce

Clone repo.

```
cd webrtc-native\godot-cpp
git checkout 7cd79e4cc64ce484ba04d1919da7d4cd39aab5ff
cd ..
scons platform=windows

```

And finally get error.

### Minimal reproduction project

_No response_

--- a/SConstruct
+++ b/SConstruct
@@ -20,9 +20,23 @@ def replace_flags(flags, replaces):
             flags[flags.index(k)] = v
 
 
+def validate_godotcpp_dir(key, val, env):
+    normalized = val if os.path.isabs(val) else os.path.join(env.Dir("#").abspath, val)
+    if not os.path.isdir(normalized):
+        raise UserError("GDExtension directory ('%s') does not exist: %s" % (key, val))
+
+
 env = Environment()
 opts = Variables(["customs.py"], ARGUMENTS)
 opts.Add(EnumVariable("godot_version", "The Godot target version", "4.1", ["3", "4.0", "4.1"]))
+opts.Add(
+    PathVariable(
+        "godot_cpp",
+        "Path to the directory containing Godot CPP folder",
+        None,
+        validate_godotcpp_dir,
+    )
+)
 opts.Update(env)
 
 # Minimum target platform versions.
@@ -33,11 +47,15 @@ if "macos_deployment_target" not in ARGUMENTS:
 if "android_api_level" not in ARGUMENTS:
     ARGUMENTS["android_api_level"] = "28"
 
+# Recent godot-cpp versions disables exceptions by default, but libdatachannel requires them.
+ARGUMENTS["disable_exceptions"] = "no"
+
 if env["godot_version"] == "3":
     if "platform" in ARGUMENTS and ARGUMENTS["platform"] == "macos":
         ARGUMENTS["platform"] = "osx"  # compatibility with old osx name
 
-    cpp_env = SConscript("godot-cpp-3.x/SConstruct")
+    sconstruct = env.get("godot_cpp", "godot-cpp-3.x") + "/SConstruct"
+    cpp_env = SConscript(sconstruct)
 
     # Patch base env
     replace_flags(
@@ -105,9 +123,18 @@ if env["godot_version"] == "3":
         for flags in (env["CCFLAGS"], env["LINKFLAGS"], cpp_env["CCFLAGS"], cpp_env["LINKFLAGS"]):
             replace_flags(flags, {"-m32": None, "-m64": None})
 elif env["godot_version"] == "4.0":
-    env = SConscript("godot-cpp-4.0/SConstruct").Clone()
+    sconstruct = env.get("godot_cpp", "godot-cpp-4.0") + "/SConstruct"
+    cpp_env = SConscript(sconstruct)
+    env = cpp_env.Clone()
 else:
-    env = SConscript("godot-cpp/SConstruct").Clone()
+    sconstruct = env.get("godot_cpp", "godot-cpp") + "/SConstruct"
+    cpp_env = SConscript(sconstruct)
+    env = cpp_env.Clone()
+
+if cpp_env.get("is_msvc", False):
+    # Make sure we don't build with static cpp on MSVC (default in recent godot-cpp versions).
+    replace_flags(env["CCFLAGS"], {"/MT": "/MD"})
+    replace_flags(cpp_env["CCFLAGS"], {"/MT": "/MD"})
 
 # Should probably go to upstream godot-cpp.
 # We let SCons build its default ENV as it includes OS-specific things which we don't

